src/number-input/__utils.ts

```diff
@@ -0,0 +1,170 @@
+import { useInterval } from "@chakra-ui/hooks";
+import { useState, useRef, useEffect, useCallback, KeyboardEvent } from "react";
+
+const FLOATING_POINT_REGEX = /^[Ee0-9\+\-\.]$/;
+
+/**
+ * Determine if a character is a DOM floating point character
+ * @see https://www.w3.org/TR/2012/WD-html-markup-20120329/datatypes.html#common.data.float
+ */
+export function isFloatingPointNumericCharacter(character: string) {
+  return FLOATING_POINT_REGEX.test(character);
+}
```

Comment:
Interesting!

Reply:
Quite nice. right. 
Only allowing the numbers to be typed and onChange.